diff --git a/libXau.def b/libXau.def
new file mode 100644
index 0000000..b681add
--- /dev/null
+++ b/libXau.def
@@ -0,0 +1,9 @@
+EXPORTS
+XauDisposeAuth
+XauFileName
+XauGetAuthByAddr
+XauGetBestAuthByAddr
+XauLockAuth
+XauReadAuth
+XauUnlockAuth
+XauWriteAuth
diff --git a/AuGetAddr.c b/AuGetAddr.c
index 9c8349f..b95f9a1 100644
--- a/AuGetAddr.c
+++ b/AuGetAddr.c
@@ -29,6 +29,9 @@ Copyright 1988, 1998  The Open Group
 #endif
 #include <X11/Xauth.h>
 #include <X11/Xos.h>
+#ifdef WIN32
+# include <io.h>
+#endif
 
 #ifdef O_CLOEXEC
 #define FOPEN_CLOEXEC "e"
diff --git a/AuGetBest.c b/AuGetBest.c
index 012479e..b1d0f96 100644
--- a/AuGetBest.c
+++ b/AuGetBest.c
@@ -29,6 +29,9 @@ Copyright 1988, 1998  The Open Group
 #endif
 #include <X11/Xauth.h>
 #include <X11/Xos.h>
+#ifdef WIN32
+# include <io.h>
+#endif
 #ifdef XTHREADS
 #include <X11/Xthreads.h>
 #endif
diff --git a/AuLock.c b/AuLock.c
index 73ed7d2..65ef25d 100644
--- a/AuLock.c
+++ b/AuLock.c
@@ -38,6 +38,7 @@ Copyright 1988, 1998  The Open Group
 #endif
 #ifdef WIN32
 # include <X11/Xwindows.h>
+# include <io.h>
 # define link rename
 #endif
 
diff --git a/meson.build b/meson.build
index dc0c34b..4842c1f 100644
--- a/meson.build
+++ b/meson.build
@@ -19,6 +19,10 @@ cc = meson.get_compiler('c')
 
 lib_args = []
 
+if host_machine.system() == 'windows'
+  lib_args += '-DWIN32'
+endif
+
 foreach f : ['explicit_bzero', 'explicit_memset', 'pathconf']
   if cc.has_function(f)
     lib_args += '-DHAVE_@0@'.format(f.to_upper())
@@ -29,15 +33,24 @@ if cc.has_header('unistd.h')
   lib_args += '-DHAVE_UNISTD_H'
 endif
 
-dep_xproto = dependency('xproto')
+if host_machine.system() == 'windows'
+  if not cc.has_header('X11/Xauth.h')
+    error('xproto headers not found')
+  endif
+  dep_xproto = declare_dependency()
+else
+  dep_xproto = dependency('xproto', version : '>= 7.0.17')
+endif
 
 if get_option('xthreads')
   lib_args += '-DXTHREADS'
   # This define is not in libXau specific code, but is part of the xproto header
   # This may be only required by HP-UX.
-  if cc.has_function('gethostbyname_r') or \
-     cc.has_function('gethostbyname_r', dependencies : cc.find_library('nls'))
-    lib_args += '-DXUSE_MTSAFE_API=1'
+  if host_machine.system() != 'windows'
+    if cc.has_function('gethostbyname_r') or \
+       cc.has_function('gethostbyname_r', dependencies : cc.find_library('nls'))
+      lib_args += '-DXUSE_MTSAFE_API=1'
+    endif
   endif
   if host_machine.system() == 'sunos'
     lib_args += ['-D_REENETRANT', '-D_POSIX_PTHREAD_SEMANTICS']
@@ -61,6 +74,7 @@ lib = library(
   dependencies : dep_xproto,
   version : '6.0.0',
   install : true,
+  vs_module_defs : 'libXau.def',
 )
 
 test(
@@ -94,4 +108,6 @@ pkg.generate(
   requires : 'xproto',
 )
 
-subdir('man')
+if host_machine.system() != 'windows'
+  subdir('man')
+endif
